packages/contracts/src/dollar/libraries/LibStaking.sol

```diff
@@ -1,532 +1,514 @@
     }
 
     /**
-     * @notice Returns current share price
-     * @return priceShare Share price
+     * @notice Sets Governance token divider param for treasury. The bigger `governanceTreasuryDivider` the less extra
+     * Governance tokens will be minted for the treasury.
+     * @notice Example: if `governanceTreasuryDivider = 5` then `100 / 5 = 20%` extra minted Governance tokens for treasury
+     * @param newGovernanceTreasuryDivider New governance divider param value
      */
-    function currentShareValue() internal view returns (uint256 priceShare) {
-        uint256 totalShares = LibChef.totalShares();
-        address stakingShareAddress = LibAppStorage
-            .appStorage()
-            .stakingShareAddress;
-        // priceShare = totalLP / totalShares
-        priceShare = LibStakingFormulas.bondPrice(
-            StakingShare(stakingShareAddress).totalLP(),
-            totalShares,
-            ONE
+    function setGovernanceTreasuryDivider(
+        uint256 newGovernanceTreasuryDivider
+    ) internal {
+        require(
+            newGovernanceTreasuryDivider > 0,
+            "Treasury divider can't be zero"
         );
+        StakingStorage storage stakingStore = stakingStorage();
+        stakingStore.governanceTreasuryDivider = newGovernanceTreasuryDivider;
     }
 
     /**
-     * @notice Updates the accumulated excess LP per share
+     * @notice Sets staking reward token
+     * @param newRewardToken New reward token address
      */
-    function _updateLpPerShare() internal {
-        address stakingShareAddress = LibAppStorage
-            .appStorage()
-            .stakingShareAddress;
-        StakingData storage ss = stakingStorage();
-        StakingShare stake = StakingShare(stakingShareAddress);
-        uint256 lpBalance = IERC20(
-            LibAppStorage.appStorage().stableSwapMetaPoolAddress
-        ).balanceOf(address(this));
-        // the excess LP is the current balance
-        // minus the total deposited LP + LP that needs to be migrated
-        uint256 totalShares = LibChef.totalShares();
-        if (
-            lpBalance >= (stake.totalLP() + ss.totalLpToMigrate) &&
-            totalShares > 0
-        ) {
-            uint256 currentLpRewards = lpBalance -
-                (stake.totalLP() + ss.totalLpToMigrate);
-
-            // is there new LP rewards to be distributed ?
-            if (currentLpRewards > ss.lpRewards) {
-                // we calculate the new accumulated LP rewards per share
-                ss.accLpRewardPerShare =
-                    ss.accLpRewardPerShare +
-                    (((currentLpRewards - ss.lpRewards) * 1e12) / totalShares);
-
-                // update the staking contract lpRewards
-                ss.lpRewards = currentLpRewards;
-            }
-        }
+    function setStakingRewardToken(address newRewardToken) internal {
+        require(newRewardToken != address(0), "Zero address detected");
```

Comment:
btw isn't it the time to use `error ZeroAddress()` instead of string errors ?

Reply:
The whole codebase uses `require` statements for error handling so it makes sense to be consistent